torch/_refs/__init__.py

```diff
@@ -2055,6 +2063,14 @@ def tensor_split(
         return tuple(splits)
 
 
+def dsplit(a: TensorLikeType, sections: DimsType) -> TensorSequenceType:
+    if a.ndim < 3:
+        raise RuntimeError(f"torch.dsplit requires a tensor with at least 3 dimension, but got a tensor with {a.ndim} dimensions!")
+    if isinstance(sections, int) and (sections == 0 or a.shape[2] % sections != 0):
```

Comment:
Should this add a check that sections is strictly positive?

Reply:
No -- we should probably introduce an error context, instead, that can produce nicer error messages without developers doing all that work.

Reply:
IIUC, #78738 tracks this error message problem in the long term.

For the scope in this PR, I think I'm striking a sweet balance here: some dsplit specific checks are done here, while relying some checks (non negative sections) in the callee.

Do we still have any blocking concern on merging this PR?

Reply:
No -- sounds great